lib/acpi/acpi.c

```diff
@@ -250,12 +247,12 @@ static ACPI_STATUS dev_resource_enum_callback(ACPI_HANDLE obj_handle, UINT32 lev
 	return status;
 }
 
-static int acpi_enum_devices(struct acpi *bus)
+static int acpi_enum_devices(void)
 {
 	LOG_DBG("");
 
 	AcpiWalkNamespace(ACPI_TYPE_DEVICE, ACPI_ROOT_OBJECT, ACPI_UINT32_MAX,
-			  dev_resource_enum_callback, NULL, bus, NULL);
+			  dev_resource_enum_callback, NULL, NULL, NULL);
```

Comment:
are just renaming "bus_ctx to simply acpi" here?

Reply:
The parameter is pa

> are just renaming "bus_ctx to simply acpi" here?

The parameter is just passes as-is to the callback. Since it's the same global variable (internal to acpi.c) the callback doesn't need it. IIRC I did update the callback implementation to not rely on the parameter anymore.

Reply:
> > are just renaming "bus_ctx to simply acpi" here?
> 
> The parameter is just passed as-is to the callback. Since it's the same global variable (internal to acpi.c) the callback doesn't need to get it as a function "user data" parameter. IIRC I did update the callback implementation to not rely on the parameter anymore.

Reply:
> are just renaming "bus_ctx to simply acpi" here?

One more clarification: it's not just a rename of the variable - we loose the name for the struct type as well, so it's not just that we don't want to pass around its pointer but that we can't.